src/util/mp_arith.h

```diff
@@ -55,7 +55,8 @@ const mp_integer binary2integer(const std::string &, bool is_signed);
 const std::string integer2bv(const mp_integer &, std::size_t width);
 
 /// convert a bit-vector representation (possibly signed) to integer
-const mp_integer bv2integer(const std::string &, bool is_signed);
+const mp_integer
+bv2integer(const std::string &, std::size_t width, bool is_signed);
```

Comment:
What is the advantage of having a ```width``` parameter here?  Is it to distinguish signed / unsigned cases?

Reply:
The meaning of a given bit-vector in two's complement does depend on the width!
Say 0xff is -1 when interpreted as 8 bit vector, but it's 255 when interpreted with 16 bits.

Reply:
It depends on it's interpretation (signed or unsigned) and width, yes.  This is fair.